src/crimson/osd/ops_executer.cc

```diff
@@ -759,6 +759,15 @@ OpsExecuter::execute_osd_op(OSDOp& osd_op)
       osd_op_params = osd_op_params_t();
       return backend.omap_set_vals(os, osd_op, txn, *osd_op_params);
     }, true);
+  case CEPH_OSD_OP_OMAPSETHEADER:
+#if 0
+    if (!pg.get_pool().info.supports_omap()) {
```

Comment:
i think this logic is correct, why commented it out?

Reply:
I was just following the template of the rest of the cases here, you can see it is commented in all of them. Do you want me to uncomment them all?

Reply:
no, i just want to know if "erasure map does not support omap" also applies to seastore and crimson. if that's true, i don't think there is a reason to comment them out.

@athanatos do you know why erasure pools do not support omap. because we cannot encode the omap KV pairs using erasure encoding when storing them? and replicate pools are much more straightforward, they just, well, replicate them, right?

Reply:
There will be bunch of such checks. Maybe it's worth to offer a dedicated `do_..._op()` variant (or make some traits machinery) to deduplicate.

Reply:
@tchaikov I think I can implement a function called maybe_do_write/read_op which will essentially have these checks inside it and also call do_write/read_op if the check passes. Something similar to what Radek suggested, what do you think?

Reply:
@AmnonHanuhov: I think we can do this clean-up in a separated PR, let's not block here.